computer_vision/src/geometry_utils.py

```diff
@@ -0,0 +1,445 @@
+    if view:
+        custom_draw_geometry_with_key_callback([pcd])
+    return pcd
+
+
+def downsample(pcd: o3d.geometry.PointCloud,
+               size: float = 0.01,
+               view: bool = False) -> o3d.geometry.PointCloud:
+    '''
+    downsample point cloud
+    :param pcd: input point cloud
+    :param size: voxel size used to downsample the point cloud
+    :param view: Add visualization
+    :return:  o3d.geometry.PointCloud
+    '''
+    downpcd = pcd.voxel_down_sample(voxel_size=size)
+    if view:
+        custom_draw_geometry_with_key_callback([downpcd])
+    return downpcd
+
+
+def cluster(pcd: o3d.geometry.PointCloud,
+            eps: float = 0.02,
+            min_points: int = 10,
+            view: bool = False) -> np.array:
+    '''
+    Cluster PointCloud using the DBSCAN algorithm and Visualization
+    :param pcd: input point cloud
+    :param eps: Density parameter that is used to find neighbouring points.
+    :param min_points: Minimum number of points to form a cluster.
+    :param view: Add visualization
+    :return: np.array
+    '''
+
+    labels = np.array(pcd.cluster_dbscan(eps, min_points))
+    if view:
+        max_label = labels.max()
+        colors = plt.get_cmap("tab20")(labels /
+                                       (max_label if max_label > 0 else 1))
+        colors[labels < 0] = 0
+        newdownpcd = o3d.geometry.PointCloud()
+        newdownpcd.points = o3d.utility.Vector3dVector(np.asarray(pcd.points))
+        newdownpcd.colors = o3d.utility.Vector3dVector(colors[:, :3])
+        custom_draw_geometry_with_key_callback([newdownpcd])
+    return labels
+
+
+def get_largest_cluster(pcd: o3d.geometry.PointCloud,
+                        labels: np.array,
+                        view: bool = False) -> o3d.geometry.PointCloud:
+    """
+    select the cluster with largest number of points --> indicate the table and bbox area
+    :param pcd: input point cloud
+    :param labels: labels obtained from cluster
+    :param view: Add visualization
+    :return: o3d.geometry.PointCloud
+    """
+    count = [(labels == i).sum() for i in np.unique(labels)]
+    max_labels = np.unique(labels)[np.argmax(count)]
+    wanted_index = np.where(labels == max_labels)[0]
+    wanted_pcd = pcd.select_down_sample(wanted_index)
+    if view:
+        custom_draw_geometry_with_key_callback([wanted_pcd])
+    return wanted_pcd
+
+
+def find_plane(pcd: o3d.geometry.PointCloud,
+               max_plane_idx: int = 2,
+               distance_threshold: float = 0.01,
+               ransac_n: int = 3,
+               num_iterations: int = 1000,
+               d_threshold: float = 0.01,
+               view: bool = False) -> [dict, dict]:
+    """
+    find plane given the point cloud
+    :param pcd: input point cloud
+    :param max_plane_idx: want to find maximun number of plane
+    :param distance_threshold: Max distance a point can be from the plane model, and still be considered an inlier.
```

Comment:
```suggestion
    :param distance_threshold: max distance a point can be from the plane model, and still be considered an inlier.
```

I will stop pointing those inconsistencies out. Just choose either or.